lib/claim.rb

```diff
@@ -0,0 +1,97 @@
+
+module Claim
+  Lock            = Mutex.new
+  @claimed_room   ||= nil
+  @last_room      ||= nil
+  @mine           ||= false
+  @buffer         = []
+  @others         = []
+  @timestamp      = Time.now
+
+  def self.claim_room(id)
+    @claimed_room = id.to_i
+    @timestamp    = Time.now
+    Log.out("claimed #{@claimed_room}", label: %i(claim room)) if defined? Log
+    Lock.unlock
+  end
+
+  def self.claimed_room
+    @claimed_room
+  end
+
+  def self.last_room
+    @last_room
+  end
+
+  def self.lock
+    Lock.lock if !Lock.owned?
+  end
+
+  def self.unlock
+    Lock.unlock if Lock.owned?
+  end
+
+  def self.current?
+    Lock.synchronize { @mine.eql?(true) }
+  end
+
+  def self.checked?(room = nil)
+    Lock.synchronize { XMLData.room_id == (room || @last_room) }
+  end
+
+  def self.info
+    info = {'Current Room' => XMLData.room_id,
+            'Mine' => Claim.mine?,
+            'Claimed Room' => Claim.claimed_room,
+            'Checked' => Claim.checked?,
+            'Last Room' => Claim.last_room,
+            'Others' => Claim.others}
+    respond JSON.pretty_generate(info)

```

Comment:
we can/should terminal table this output

Reply:
This is used for primarily debugging/info purposes, so I'm ok with adding description column.